assets/js/translator.js

```diff
@@ -350,13 +339,391 @@ if(modeEnabled('translation')) {
             persistChoices('translator');
         });
 
+        initChainGraph();
+        refreshChosenPath();
         setupLanguageSelectors();
         setupTextTranslation();
         setupWebpageTranslation();
         setupDocTranslation();
     });
 }
 
+function updatePairList() {
+    pairs = $('input#chainedTranslation').prop('checked') ? chainedPairs : originalPairs;
+}
+
+function getShortestChainedPaths() {
+    var paths = {};
+    $.each(srcLangs, function (i, src) {
+        var langs = filterLangList(srcLangs.concat(dstLangs).filter(onlyUnique));
+        var dists = {};
+        dists[src] = 0;
+        dists.dummy = Number.MAX_SAFE_INTEGER;
+        var prevs = {};
+
+        while(langs.length > 0) {
+            var mini = 'dummy';
+            $.each(langs, function (i, lang) {
+                if((dists[lang] !== undefined) && (dists[lang] < dists[mini])) {
+                    mini = lang;
+                }
+            });
+            langs.splice(langs.indexOf(mini), 1);
+            if(pairs[mini]) {
+                $.each(pairs[mini], function (i, trgt) {
+                    if(langs.indexOf(trgt) !== -1) {
+                        var newdist;
+                        if(dists[mini] === undefined) {
+                            dists[mini] = Number.MAX_SAFE_INTEGER;
+                            newdist = Number.MAX_SAFE_INTEGER;
+                        }
+                        else {
+                            newdist = dists[mini] + weights[mini][trgt];
+                        }
+                        if((dists[trgt] === undefined) || (dists[mini] + weights[mini][trgt] < dists[trgt])) {
+                            dists[trgt] = newdist;
+                            prevs[trgt] = mini;
+                        }
+                    }
+                });
+            }
+        }
+
+        paths[src] = {};
+        $.each(Object.keys(prevs), function (i, trgt) {
+            var prev = trgt;
+            var path = [trgt];
+            while(prevs[prev]) {
+                prev = prevs[prev];
+                path.push(prev);
+            }
+            path.reverse();
+            paths[src][trgt] = {'path': path, 'weight': getWeight(path)};
+        });
+    });
+
+    $.each(srcLangs, function (i, src) {
+        chainedPairs[src] = Object.keys(paths[src]);
+    });
+
+    return paths;
+    // chainedPaths[src][dst] = [{path: [a, b, c], weight: 2}, {path: [a, d, e, c], weight: 3}]
```

Comment:
:) (dead code!)

Reply:
This was actually supposed to be the "brief comment" you mentioned in the previous comment on line 355 :) I'll move it up there.